lib/twitter_cldr/shared/likely_subtags.rb

```diff
@@ -0,0 +1,106 @@
+# encoding: UTF-8
+
+# Copyright 2012 Twitter, Inc
+# http://www.apache.org/licenses/LICENSE-2.0
+
+module TwitterCldr
+  module Shared
+
+    class UnrecognizedSubtagsError < StandardError; end
+
+    class LikelySubtags
+      class << self
+        def locale_for(locale_text)
+          locale = Locale.parse(locale_text)
+          lookup(locale)
+        end
+
+        protected
+
+        # Try each of the following in order (where the fields exist). The
+        # notation field³ means field¹ if it exists, otherwise field².
+        #
+        # 1. Lookup language¹ _ script¹ _ region¹. If in the table, return the
+        #    language² _ script² _ region² + variants.
+        #
+        # 2. Lookup language¹ _ script¹. If in the table, return language² _
+        #    script² _ region³ + variants.
+        #
+        # 3. Lookup language¹ _ region¹. If in the table, return language² _
+        #    script³ _ region² + variants.
+        #
+        # 4. Lookup language¹. If in the table, return language² _ script³ _
+        #    region³ + variants.
+        def lookup(locale)
+          first_lookup(locale)
```

Comment:
Do you think we could do

```ruby
first_lookup(locale) ||
  second_lookup(locale) ||
  third_lookup(locale) ||
  fourth_lookup(locale)
```
instead of making these methods call each other? It'd keep them separated and might make it easier to test/debug them.